ecosystem/typescript/sdk/src/generated/core/request.ts

```diff
@@ -12,6 +12,111 @@ import { CancelablePromise } from './CancelablePromise';
 import type { OnCancel } from './CancelablePromise';
 import type { OpenAPIConfig } from './OpenAPI';
 
+interface Cookie {
+  name: string;
+  value: string;
+  expires?: Date;
+  path?: string;
+  sameSite?: "Lax" | "None" | "Strict";
+  secure?: boolean;
+}
+
+class CookieJar {
+  constructor(private jar = new Map<string, Cookie[]>()) {}
+
+  setCookie(url: URL, cookieStr: string) {
+    const key = url.origin.toLowerCase();
+    if (!this.jar.has(key)) {
+      this.jar.set(key, []);
+    }
+
+    const cookie = CookieJar.parse(cookieStr);
+    this.jar.set(key, [...(this.jar.get(key)?.filter((c) => c.name !== cookie.name) || []), cookie]);
+  }
+
+  getCookies(url: URL): Cookie[] {
+    const key = url.origin.toLowerCase();
+    if (!this.jar.get(key)) {
+      return [];
+    }
+
+    // Filter out expired cookies
+    return this.jar.get(key)?.filter((cookie) => !cookie.expires || cookie.expires > new Date()) || [];
+  }
+
+  static parse(str: string): Cookie {
+    if (typeof str !== "string") {
+      throw new Error("argument str must be a string");
+    }
+
+    const parts = str.split(";").map((part) => part.trim());
```

Comment:
? what does this look like just curious, `;` feels like a weird delimeter to split on

Reply:
I think it's the standard delimiter for the Cookie http header